cli/waiter/subcommands/ping.py

```diff
@@ -1,87 +1,77 @@
 import logging
-import threading
-import time
-from enum import Enum
 
 from waiter import http_util, terminal
 from waiter.format import format_status
 from waiter.querying import query_token, print_no_data, query_service, get_services_using_token, get_service
-from waiter.util import guard_no_cluster, response_message, print_error, check_positive, wait_until, is_service_current
+from waiter.util import guard_no_cluster, print_error, check_positive, wait_until, is_service_current, response_message
 
 
-def ping_on_cluster(cluster, health_check_endpoint, timeout, wait_for_request, token_name, service_exists_fn):
+def ping_on_cluster(cluster, timeout, wait_for_request, token_name, service_exists_fn):
     """Pings using the given token name (or ^SERVICE-ID#) in the given cluster."""
-
-    class PingResult(Enum):
-        UNKNOWN = 1
-        SUCCESS = 2
-        FAILURE = 3
-        ERRORED = 4
-
-        def succeeded(self):
-            return self is PingResult.SUCCESS
-
-        def failed(self):
-            return self is PingResult.FAILURE
-
     cluster_name = cluster['name']
-    ping_result = PingResult.UNKNOWN
 
     def perform_ping():
-        nonlocal ping_result
+        status = None
         try:
+            timeout_seconds = timeout if wait_for_request else 5
             headers = {
                 'X-Waiter-Token': token_name,
-                'X-Waiter-Fallback-Period-Secs': '0'
+                'X-Waiter-Timeout': str(timeout_seconds * 1000)
             }
-            read_timeout = timeout if wait_for_request else 5
-            resp = http_util.get(cluster, health_check_endpoint, headers=headers, read_timeout=read_timeout)
+            read_timeout = timeout_seconds * 2
+            resp = http_util.get(cluster, '/waiter-ping', headers=headers, read_timeout=read_timeout)
             logging.debug(f'Response status code: {resp.status_code}')
+            resp_json = resp.json()
             if resp.status_code == 200:
-                print(terminal.success(f'Ping successful.'))
-                print(resp.text)
-                ping_result = PingResult.SUCCESS
+                status = resp_json['service-state']['status']
+                ping_response = resp_json['ping-response']
+                ping_response_result = ping_response['result']
+                if ping_response_result == 'received-response':
+                    ping_response_status = ping_response['status']
+                    if ping_response_status == 200:
+                        print(terminal.success('Ping successful.'))
```

Comment:
Should we further qualify `Ping` as `Health check ping`? (This applies to the uses of ping in the ping response if-else branch)

Reply:
I'm not sure if that's more clear -- what's your thinking here?